modules/ROOT/pages/basic-authentication-ldap-concept.adoc

```diff
@@ -7,42 +7,76 @@ endif::[]
 |===
 >s| Policy Name | Basic Authentication: LDAP
 >s|Summary      | Allows access based on the basic authorization mechanism, with user-password defined on LDAP
->s|Category |
+>s|Category | Security
 >s|First Mule version available | 4.0.0
 .1+>.^s| Returned Status Codes
-|
+|401 - Unauthorized or invalid client application credentials
 |===
 
-The Lightweight Directory Access Protocol (LDAP) authentication policy establishes the configuration details for an Open LDAP or Active Directory LDAP that you set up for your enterprise. This policy is available only to an API you are managing in Mule 4 or later.
+The Lightweight Directory Access Protocol (LDAP) authentication policy specifies how an API with an Open LDAP or Active Directory LDAP mechanism must be authenticated. This policy is available only in Mule 4 or later. 
 
-You can follow the general instructions for applying a policy, including those for applying the policy to resources and methods.
+When an error is encountered, the policy returns an "WWW-Authenticate" HTTP header field using the following format:`WWW-Authenticate: Basic realm="mule-realm"`.
 
-The following properties configure the LDAP policy for connecting to your LDAP or Active Directory.
 
-* LDAP Server URL
-+
-The URL of your LDAP server, including the port number.
-+
-* LDAP User DN
-+
-The name of the user or user group with access to traverse and list users of LDAP.
-+
-* LDAP User Password
-+
-The password for the user or user group.
-+
-* LDAP Search Base
-+
-The starting point for the search in the directory tree.
-+
-* LDAP Search Filter
+== How This Policy Works
+
+//Nahuel can we explain here briefly how this policy works and then explain to the users how to the search scope works.
```

Comment:
The policy intercepts the request to the protected resource and looks for the "Authorization" HTTP header. Then it extracts its associated value which is in fact a username and password encoded in Base64. When that is done, the policy requests the configured LDAP instance to determine if the user/password are correct in the provided LDAP context1. You have two ways of doing this:
- Look for valid user/password pair in the level specified
- Look for valid user/password pair in the level specified or below (comprehensive but much slower)

1: LDAP is kind of inverted tree, where each leave is essentially a username/password pair with some associated metadata. Each level is like a tree branch. Explaining what LDAP is and how it works is out of the scope for this document. More info on: https://en.wikipedia.org/wiki/Lightweight_Directory_Access_Protocol.

Reply:
Awesome! Thanks Nahuel! Done1